packages/@react-spectrum/dnd/src/useDnDHooks.ts

```diff
@@ -0,0 +1,96 @@
+ * governing permissions and limitations under the License.
+ */
+
+import {
+  DraggableCollectionOptions,
+  DraggableCollectionState,
+  DroppableCollectionState,
+  DroppableCollectionStateOptions,
+  useDraggableCollectionState,
+  useDroppableCollectionState
+} from '@react-stately/dnd';
+import {DraggableCollectionProps, DragItem} from '@react-types/shared';
+import {
+  DraggableItemProps,
+  DraggableItemResult,
+  DragPreview,
+  DropIndicatorAria,
+  DropIndicatorProps,
+  DroppableCollectionOptions,
+  DroppableCollectionResult,
+  DroppableItemOptions,
+  DroppableItemResult,
+  useDraggableItem,
+  useDropIndicator,
+  useDroppableCollection,
+  useDroppableItem
+} from '@react-aria/dnd';
+import {DroppableCollectionProps} from '@react-types/shared';
+import {Key, RefObject, useMemo} from 'react';
+
+export interface DragHooks {
+  useDraggableCollectionState(props: Omit<DraggableCollectionOptions, 'getItems'>): DraggableCollectionState,
+  useDraggableItem(props: DraggableItemProps, state: DraggableCollectionState): DraggableItemResult,
+  DragPreview: typeof DragPreview
+}
+
+export interface DropHooks {
+  useDroppableCollectionState(props: DroppableCollectionStateOptions): DroppableCollectionState,
+  useDroppableCollection(props: DroppableCollectionOptions, state: DroppableCollectionState, ref: RefObject<HTMLElement>): DroppableCollectionResult,
+  useDroppableItem(options: DroppableItemOptions, state: DroppableCollectionState, ref: RefObject<HTMLElement>): DroppableItemResult,
+  useDropIndicator(props: DropIndicatorProps, state: DroppableCollectionState, ref: RefObject<HTMLElement>): DropIndicatorAria
+}
+
+export interface DnDHooks {
+  dragHooks: DragHooks,
+  dropHooks: DropHooks
+}
+
+export interface DnDOptions extends Omit<DraggableCollectionProps, 'preview' | 'getItems'>, DroppableCollectionProps {
+  /**
+   * A function that returns the items being dragged. If not specified, assumes that the collection is not draggable.
+   * @default () => []
+   */
+  getItems?: (keys: Set<Key>) => DragItem[]
+}
+
+export function useDnDHooks(options: DnDOptions): DnDHooks {
+  let dragHooks = useMemo(() => ({
+    useDraggableCollectionState(props: DraggableCollectionOptions) {
+      return useDraggableCollectionState({
+        getItems: () => [],
+        ...props,
+        ...options
+      });
+    },
+    useDraggableItem,
+    DragPreview
+  }), [options]);
+
+  let dropHooks = useMemo(() => ({
+    useDroppableCollectionState(props) {
+      return useDroppableCollectionState({...props, ...options});
+    },
+    useDroppableItem,
+    useDroppableCollection(props, state, ref) {
+      return useDroppableCollection({...props, ...options}, state, ref);
+    },
+    useDropIndicator
+  }), [options]);
+
+
+  // If the user doesn't want their collection to be draggable/droppable, they can just not pass dragHooks/dropHooks to their collection
+  return {
+    dragHooks,
+    dropHooks
```

Comment:
Don't return separate hooks, just return one merged one, then update ListView logic
```
  let isListDraggable = !!dragHooks;
  let isListDroppable = !!dropHooks;
```
and drill into the one prop that is passed to ListView

Reply:
On second thought, we would then need to conditionally include/exclude the drag or drop hooks depending on the options the user passes, which would involve parsing all of the options. Keep as is for now until feedback